tests/scenarios/helpers/command-watcher.ts

```diff
@@ -0,0 +1,143 @@
+import execa, { type Options, type ExecaChildProcess } from 'execa';
+import path from 'path';
+
+export const DEFAULT_TIMEOUT = process.env.CI ? 90000 : 30000;
+
+export default class CommandWatcher {
+  static launch(command: string, args: readonly string[], options: Options<string> = {}): CommandWatcher {
+    return new CommandWatcher(
+      execa(path.join(options.cwd as string, 'node_modules/.bin', command), [...args], {
+        ...options,
+        all: true,
+      })
+    );
+  }
+
+  private lines: string[] = [];
+  private nextWaitedLine = 0;
+  private exitCode: number | null = null;
+  private currentWaiter: (() => void) | undefined;
+
+  constructor(private process: ExecaChildProcess) {
+    process.all!.on('data', data => {
+      const lines = data.toString().split(/\r?\n/);
+      this.lines.push(...lines);
+      this.currentWaiter?.();
+    });
+
+    process.on('exit', code => {
+      // TODO why can code be null here?
+      this.exitCode = code ?? 0;
+      this.currentWaiter?.();
+    });
+  }
+
+  private async internalWait(timedOut?: Promise<void>): Promise<void> {
+    if (this.currentWaiter) {
+      throw new Error(`bug: only one wait at a time`);
+    }
+    try {
+      await Promise.race(
+        [
+          timedOut,
+          new Promise<void>(resolve => {
+            this.currentWaiter = resolve;
+          }),
+        ].filter(Boolean)
+      );
+    } finally {
+      this.currentWaiter = undefined;
+    }
+  }
+
+  // A method inspired by strip-ansi and ansi-regex packages,
+  // implemented manually here to workaround issues with
+  // ts-nodes and ES modules imports
+  private stripAnsi(string: string): string {
+    const pattern = [
+      '[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]+)*|[a-zA-Z\\d]+(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)',
+      '(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-nq-uy=><~]))',
+    ].join('|');
+    let ansiRegExp = new RegExp(pattern, 'g');
+    return string.replace(ansiRegExp, '');
+  }
```

Comment:
I don't like doing this 🙈 this is a little function that basically re-implements `strip-ansi` and `ansi-regex` packages. 

It's a very little piece of code so I did this after failing to import the `strip-ansi` module into `CommandWatcher`. I had the same error `Unknown file extension ".ts"` as I had when using `node-fetch` > 3, and after trying a few things it was rather:
```
Error [ERR_REQUIRE_ESM]: require() of ES Module 
.../embroider/node_modules/.pnpm/strip-ansi@7.1.0/node_modules/strip-ansi/index.js 
from 
.../embroider/tests/scenarios/helpers/command-watcher.js 
not supported.
Instead change the require of index.js in .../embroider/tests/scenarios/helpers/command-watcher.js 
to a dynamic import() which is available in all CommonJS modules.
at Object.require.extensions.<computed> [as .js] (.../embroider/node_modules/.pnpm/ts-node@10.9.2_typescript@5.2.2/node_modules/ts-node/dist/index.js:851:20)
```
☝️ I don't know how to debug these `ts-node`-related errors.